BEST_PRACTICES.md

```diff
@@ -0,0 +1,88 @@
+# Introduction
+This is a collection of tips, advice, gotchas and other best practices
+for using the [Jenkins Workflow
+plugin](https://github.com/jenkinsci/workflow-plugin/blob/master/README.md). Contributions
+and comments are happily accepted.
+
+# General tips and advice
+* Do everything that makes sense there within `stage`s. This will make your builds
+easier to visualize, debug, etc.
+* Do all real work that involves shelling out, building, etc, within
```

Comment:
*shelling out* may be a problem for an international audience. I'm not entirely sure I understand it either.